src/platform/magento2/order/orderProcessor.js

```diff
@@ -0,0 +1,302 @@
+    const orderIncrementId = order.increment_id;
+    logger.debug(`${THREAD_ID} [OK] Order increment id`, orderIncrementId);
+    updateStep(job);
+  } catch (err) {
+    logger.warn('could not fetch increment_id for Order', err, typeof err);
+  }
+}
+
+/**
+ * @param {json} orderData order data in format as described in '../models/order.md'
+ * @param job
+ * @returns {Promise<*>}
+ */
+async function updateCartItems ({ orderData, job }) {
+  const {
+    cart_id: cartId,
+    products: clientItems
+  } = orderData;
+
+  const THREAD_ID = orderUtilities.getThreadId(job);
+  logger.info(THREAD_ID + '< Cart Id', cartId);
+  const api = apiUtilities.getApi(order.store_code);
+
+  const isThisAuthOrder = orderUtilities.isThisAuthOrder(orderData);
+  const serverItems = await api.cart.pull(null, cartId, null, isThisAuthOrder);
+  var syncPromises = [];
+
+  for (const clientItem of clientItems) {
+    const serverItem = serverItems.find(itm => productsEquals(itm, clientItem));
+    if (!serverItem) {
+      logger.info(THREAD_ID + '< No server item for ' + clientItem.sku)
+      syncPromises.push(api.cart.update(null, cartId, { // use magento API
+        sku: clientItem.parentSku && config.cart.setConfigurableProductOptions ? clientItem.parentSku : clientItem.sku,
+        qty: clientItem.qty,
+        product_option: clientItem.product_option,
+        quote_id: cartId
+      }, isThisAuthOrder));
+    } else if (serverItem.qty !== clientItem.qty) {
+      logger.info(THREAD_ID + '< Wrong qty for ' + clientItem.sku, clientItem.qty, serverItem.qty)
+      syncPromises.push(api.cart.update(null, cartId, { // use magento API
+        sku: clientItem.parentSku && config.cart.setConfigurableProductOptions ? clientItem.parentSku : clientItem.sku,
+        qty: clientItem.qty,
+        product_option: clientItem.product_option,
+        item_id: serverItem.item_id,
+        quote_id: cartId
+      }, isThisAuthOrder));
+    } else {
+      // here we need just update local item_id
+      logger.info(THREAD_ID + '< Server and client items synced for ' + clientItem.sku);
+    }
+  }
+
+  for (const serverItem of serverItems) {
+    if (serverItem) {
+      const clientItem = clientItems.find(itm => productsEquals(itm, serverItem));
+
+      if (!clientItem) {
+        logger.info(THREAD_ID + '< No client item for ' + serverItem.sku + ', removing from server cart') // use magento API
+        syncPromises.push(api.cart.delete(null, cartId, { // delete server side item if not present if client's cart
+          sku: serverItem.sku,
+          item_id: serverItem.item_id
+        }, isThisAuthOrder))
+      }
+    }
+  }
+
+  const results = await Promise.all(syncPromises);
+  updateStep(job);
+  logger.info(THREAD_ID + '< Server cart in sync')
+  logger.debug(THREAD_ID + results);
+}
+
+async function placeOrder ({ orderData, shippingInformation, job }) {
+  const {
+    cart_id: cartId
+  } = orderData;
+
+  var {
+    payment_methods: [
```

Comment:
Please use const or let. What's more - that struct looks weird - I think it won't work or will be at least confusing - please do not double+ nest while using spread

Reply:
It's just object and array destructuring, that's how it works.